### PR TITLE
Removes Web3 from the deploy script

### PR DESCRIPTION
**Pull request by ajsantander**

### CODE REVIEW

**Review by leomassazza — APPROVED** [review]
LGTM

File: publish/src/Deployer.js [hunk 1]
```diff
@@ -121,41 +106,38 @@ class Deployer {
...
 			value: 0,
 			gasPrice: ethers.utils.parseUnits(this.gasPrice.toString(), 'gwei'),
 		};
 
-		if (this.useOvm) {
-			tx.gas = await this.provider.web3.eth.estimateGas(tx);
-		} else {
-			tx.gas = 1000000;
+		if (!this.useOvm) {
+			tx.gasLimit = 1000000;
```
Comment (justinjmoses) [inline publish/src/Deployer.js:121]: why remove this?

Comment (justinjmoses) [reply publish/src/Deployer.js:121]: ah because of below with the override gas limit

Comment (justinjmoses) [reply publish/src/Deployer.js:121]: then perhaps this isn't even needed?

File: publish/src/command-utils/transact.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ const performTransactionalStep = async ({
 	writeArg, // none, 1 or an array of args, array will be spread into params
 	gasLimit,
 	gasPrice,
+	generateSolidity,
```
Comment (justinjmoses) [inline publish/src/command-utils/transact.js:24]: was this lost before somehow in a merge?

**Review by justinjmoses — APPROVED** [review]
just a couple of comments and a question

File: publish/src/command-utils/transact.js (hunk 2 above)
Comment (ajsantander) [reply publish/src/command-utils/transact.js:24]: Yeah, we renamed `performTransactionalStep` to `performTransactionalStepWeb3` and introduced a new `performTransactionalStep` in transact.js, which some non-deploy scripts (already ported to ethers) where using.

The two functions came out of sync, and I manually compared them and introduced these changes.

File: publish/src/Deployer.js (hunk 1 above)
Comment (ajsantander) [reply publish/src/Deployer.js:121]: Will review and handle in a separate PR.

**Pull request merged**